pkg/dockerclient/shell.go

```diff
@@ -554,3 +512,64 @@ func (c *ShellClient) getLastNonEmptyLine(lines []string, offset int) string {
 func (c *ShellClient) replaceSingleQuotes(input string) string {
 	return strings.Replace(input, "'", "’", -1)
 }
+
+func (c *ShellClient) resolveDockerBuildNetwork() string {
+
+	// may contain none as a value
+	networkInterface := os.Getenv("NUCLIO_DOCKER_BUILD_NETWORK")
+	if networkInterface == "" {
+		networkInterface = common.GetEnvOrDefaultString("NUCLIO_BUILD_USE_HOST_NET", "host")
+	}
+	switch networkInterface {
+	case "host":
+		fallthrough
+	case "default":
+		fallthrough
+	case "none":
+		return fmt.Sprintf("--network %s", networkInterface)
+	default:
+		return ""
+	}
+}
+
+func (c *ShellClient) build(buildOptions *BuildOptions, buildArgs string, cacheOption string) error {
+	var err error
```

Comment:
```suggestion
	var lastBuildErr error
```

Giving this a distinct name and naming the internal error in `RetryUntilSuccessful` "err" is more familiar